app/controllers/alaveteli_pro/plans_controller.rb

```diff
@@ -22,9 +22,19 @@ def authenticate
   end
 
   def check_existing_subscription
-    if @user.pro_account
-      flash[:error] = _('You are already subscribed to this plan')
-      redirect_to alaveteli_pro_dashboard_path
+    customer_id = @user.pro_account.try(:stripe_customer_id)
+
+    if customer_id
+      customer = Stripe::Customer.retrieve(customer_id)
+      # TODO: This doesn't take the plan in to account
+      active_subscriptions =
+        customer.subscriptions.map(&:status).
+          any? { |status| status != 'canceled' }
```

Comment:
Actually a 'canceled' subscription is still "active" until the end of the billing period, so we want to block if `customer.subscriptions.any?`

Reply:
This _should_ change to `customer.subscriptions.any?`, but a subscription with a `'canceled'` state is just so that you have the object available after you delete the subscription. When you reload a customer after deleting their subscription, `customer.subscriptions.any?` is `false`